cmd/opm/generate/cmd.go

```diff
@@ -83,6 +85,7 @@ value of each duplicate key will be added to the generated Dockerfile.
 		},
 	}
 	cmd.Flags().StringVarP(&baseImage, "binary-image", "i", containertools.DefaultBinarySourceImage, "Image in which to build catalog.")
+	cmd.Flags().BoolVarP(&lite, "lite", "t", false, "Generate a smaller, binary-less Dockerfile.")
```

Comment:
What do you think of something like `--include-binary=false` to generate the binary-less Dockerfile? To maintain existing functionality it could default to `--include-binary=true`

Reply:
It's analogous to what I modeled, so I'm fine with it.  I expect there will be a few opinions on this one, so I'm waiting until we get a few before we let them battle royale.
😁

Reply:
I'm not strongly opinionated but just thought I'd make a suggestion for another name since you mentioned folks might not like the `--lite` flag :) - I actually don't mind it so whatever floats your boat here is fine by me

Reply:
Yeah, `lite` is living on borrowed time. 😄  
We have to effectively communicate that we're both
1. deciding whether or not to use a builder stage; AND 
2. specifying a base image

Reply:
I'm still interested in the option of `--binary-image=scratch` (and we could beef up the CLI help text slightly to explain the pairing with CatalogSource's `spec.grpcPodConfig.extractContent`).

AND I want to again discuss the validity of changing the Dockerfile template to just _always_ be multi-stage.  If you squint, you could call it a bug that `--binary-image=scratch` results in a broken Dockerfile and it should have been multi-stage from the beginning.

For users out there that generate the Dockerfile and then manipulate it afterward, my suggestion in retrospect would be: don't. If you want to modify the Dockerfile after generation, you are better off just maintaining your own Dockerfile.

Long-term I want to eliminate this command and build a better tool that can build and push images without a container runtime, so IMO this is living on borrowed time.

Reply:
> AND I want to again discuss the validity of changing the Dockerfile template to just always be multi-stage. If you squint, you could call it a bug that --binary-image=scratch results in a broken Dockerfile and it should have been multi-stage from the beginning.

If --binary-image=scratch is the way to indicate that we are building a binless image, how does a user ask for a different builder image?

I'm pushing a new commit to switch to a two-flag approach to reduce ambiguity.  I think maybe we can still get to single-template-land though.

Reply:
> If --binary-image=scratch is the way to indicate that we are building a binless image, how does a user ask for a different builder image?

Great point!